api/v1alpha1/hostdevicenetwork_types.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright 2021 NVIDIA
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	HostDeviceNetworkCRDName = "HostDeviceNetwork"
+)
+
+// HostDeviceNetworkSpec defines the desired state of HostDeviceNetwork
+type HostDeviceNetworkSpec struct {
+	// Namespace of the NetworkAttachmentDefinition custom resource
+	NetworkNamespace string `json:"networkNamespace,omitempty"`
```

Comment:
the user may deploy the operator with multiple host device resources configured in device plugin.
each resource may then map to a different network.

This CRD should have a `ResourceName` field

Reply:
@e0ne  can you add the ResourceName per of adrian comments